File: test/corpus/basic/04-deterministic-operations/case.go
package main

import (
	"fmt"
	colorful "github.com/lucasb-eyer/go-colorful"
)

func main() {
	fmt.Printf("=== Deterministic Color Operations Test ===\n")
	
	// Test 1: Direct HSV creation with known values
	c1 := colorful.Hsv(0, 0.75, 0.55)  // Same as FastWarmPalette logic but deterministic
	fmt.Printf("HSV(0, 0.75, 0.55): R=%.6f G=%.6f B=%.6f Hex=%s\n", c1.R, c1.G, c1.B, c1.Hex())
	
	c2 := colorful.Hsv(120, 0.65, 0.45)
	fmt.Printf("HSV(120, 0.65, 0.45): R=%.6f G=%.6f B=%.6f Hex=%s\n", c2.R, c2.G, c2.B, c2.Hex())
	
	c3 := colorful.Hsv(240, 0.70, 0.50)
	fmt.Printf("HSV(240, 0.70, 0.50): R=%.6f G=%.6f B=%.6f Hex=%s\n", c3.R, c3.G, c3.B, c3.Hex())
	
	// Test 2: Color space conversions
	rgb := colorful.Color{R: 0.8, G: 0.4, B: 0.2}
	h, s, v := rgb.Hsv()
	fmt.Printf("RGB(0.8,0.4,0.2) -> HSV: %.6f %.6f %.6f\n", h, s, v)
	
	l, a, b := rgb.Lab()
	fmt.Printf("RGB(0.8,0.4,0.2) -> Lab: %.6f %.6f %.6f\n", l, a, b)
	
	// Test 3: Manual FastWarmPalette-style generation with known values
	fmt.Printf("\n=== Manual Warm Palette Generation (Deterministic) ===\n")
	count := 3
	for i := 0; i < count; i++ {
		hue := float64(i) * (360.0 / float64(count))
		sat := 0.65  // Fixed saturation instead of random
		val := 0.45  // Fixed value instead of random
		c := colorful.Hsv(hue, sat, val)
		fmt.Printf("Color %d: R=%.6f G=%.6f B=%.6f Hex=%s (H=%.1f S=%.2f V=%.2f)\n", 
			i, c.R, c.G, c.B, c.Hex(), hue, sat, val)
	}
}

File: test/corpus/basic/04-deterministic-operations/case.ts
import * as GoStyle from '../../../../src/go-style';

console.log("=== Deterministic Color Operations Test ===");

// Test 1: Direct HSV creation with known values
const c1 = GoStyle.Hsv(0, 0.75, 0.55);  // Same as FastWarmPalette logic but deterministic
console.log(`HSV(0, 0.75, 0.55): R=${c1.R.toFixed(6)} G=${c1.G.toFixed(6)} B=${c1.B.toFixed(6)} Hex=${c1.Hex()}`);

const c2 = GoStyle.Hsv(120, 0.65, 0.45);
console.log(`HSV(120, 0.65, 0.45): R=${c2.R.toFixed(6)} G=${c2.G.toFixed(6)} B=${c2.B.toFixed(6)} Hex=${c2.Hex()}`);

const c3 = GoStyle.Hsv(240, 0.70, 0.50);
console.log(`HSV(240, 0.70, 0.50): R=${c3.R.toFixed(6)} G=${c3.G.toFixed(6)} B=${c3.B.toFixed(6)} Hex=${c3.Hex()}`);

// Test 2: Color space conversions
const rgb = new GoStyle.Color(0.8, 0.4, 0.2);
const [h, s, v] = rgb.Hsv();
console.log(`RGB(0.8,0.4,0.2) -> HSV: ${h.toFixed(6)} ${s.toFixed(6)} ${v.toFixed(6)}`);

const [l, a, b] = rgb.Lab();
console.log(`RGB(0.8,0.4,0.2) -> Lab: ${l.toFixed(6)} ${a.toFixed(6)} ${b.toFixed(6)}`);

// Test 3: Manual FastWarmPalette-style generation with known values
console.log("\n=== Manual Warm Palette Generation (Deterministic) ===");
const count = 3;
for (let i = 0; i < count; i++) {
  const hue = i * (360.0 / count);
  const sat = 0.65;  // Fixed saturation instead of random
  const val = 0.45;  // Fixed value instead of random
  const c = GoStyle.Hsv(hue, sat, val);
  console.log(`Color ${i}: R=${c.R.toFixed(6)} G=${c.G.toFixed(6)} B=${c.B.toFixed(6)} Hex=${c.Hex()} (H=${hue.toFixed(1)} S=${sat.toFixed(2)} V=${val.toFixed(2)})`);
}